### PR TITLE
Execute page added and modified which work in both testsuite and test…

### PR DESCRIPTION
**Pull request by tanmaythole**
…case execution

### CODE REVIEW

File: src/Components/ExecuteForm/index.jsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+const ExecuteForm = ({
+  label,
+  data,
+  environments,
+  isEnvsLoading,
+  handleExecute,
+}) => {
+  // This component and its methods need to be change with respect to FormSelect component
+  const [options, setOptions] = useState([]);
+  const [selectedEnv, setSelectedEnv] = useState();
```
Comment (ChinmaySonawane) [inline src/Components/ExecuteForm/index.jsx]: don't we  any default value to `selectedEnv`

File: src/Components/ExecuteForm/index.jsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  const handleEnvChange = (_name, value) => {
+    setSelectedEnv(value);
+  };
+
+  const onExecute = () => {
+    handleExecute(data.id, selectedEnv);
+  };
+
+  return (
+    !isEnvsLoading && (
```
Comment (ChinmaySonawane) [inline src/Components/ExecuteForm/index.jsx]: we can use loader here for better user experience

File: src/Components/ExecuteForm/index.jsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    )
+  );
+};
+
+ExecuteForm.propTypes = {
+  data: PropTypes.object.isRequired,
+  environments: PropTypes.array.isRequired,
+  isEnvsLoading: PropTypes.bool,
+  handleExecute: PropTypes.func.isRequired,
+  label: PropTypes.string,
```
Comment (ChinmaySonawane) [inline src/Components/ExecuteForm/index.jsx]: This should be also isRequired

File: src/Components/ProjectsComponent/ExecuteComponent/index.jsx [hunk 4]
```diff
@@ -1,65 +1,56 @@
...
+
+      default:
+        return "";
+    }
+  };
+
+  return (
+    <Card className={accordionItemClass(status)} body>
+      <Link
+        to={isExecute ? `/project/${project}/reports/${reportId}` : ""}
```
Comment (ChinmaySonawane) [inline src/Components/ProjectsComponent/ExecuteComponent/index.jsx]: For all links or url, we can get it in variable and then pass variable.

File: src/containers/ProjectContainer/TestcaseContainer/index.jsx [hunk 5]
```diff
@@ -1,203 +1,235 @@
...
+        level: "testcase",
+      })
+    );
+    navigate(`/project/${projectName}/execute/${testcase}`);
+  };
+
+  const onRemoveSelectedTeststep = (index) => {
+    setSelectedTeststeps((prevState) => {
+      let fields = prevState.filter(function (_value, ind) {
+        return ind !== index;
```
Comment (ChinmaySonawane) [inline src/containers/ProjectContainer/TestcaseContainer/index.jsx]: For this you can rename `index` to `selectedIndex` and `ind` to  `index` .

File: src/containers/ProjectContainer/TestcaseContainer/index.jsx [hunk 6]
```diff
@@ -1,203 +1,235 @@
...
+      });
+      return fields;
+    });
+  };
+
+  const onSelectedTeststepsChange = (_name, value) => {
```
Comment (ChinmaySonawane) [inline src/containers/ProjectContainer/TestcaseContainer/index.jsx]: if you not using name variable in function them remove it

File: src/Components/ExecuteForm/index.jsx (hunk 1 above)
Comment (tanmaythole) [reply src/Components/ExecuteForm/index.jsx]: done 👍🏻

File: src/Components/ExecuteForm/index.jsx (hunk 2 above)
Comment (tanmaythole) [reply src/Components/ExecuteForm/index.jsx]: done... loader component added 👍🏻

File: src/Components/ExecuteForm/index.jsx (hunk 3 above)
Comment (tanmaythole) [reply src/Components/ExecuteForm/index.jsx]: done 👍🏻

File: src/Components/ProjectsComponent/ExecuteComponent/index.jsx (hunk 4 above)
Comment (tanmaythole) [reply src/Components/ProjectsComponent/ExecuteComponent/index.jsx]: done 👍🏻

File: src/containers/ProjectContainer/TestcaseContainer/index.jsx (hunk 5 above)
Comment (tanmaythole) [reply src/containers/ProjectContainer/TestcaseContainer/index.jsx]: done 👍🏻

File: src/containers/ProjectContainer/TestcaseContainer/index.jsx (hunk 6 above)
Comment (tanmaythole) [reply src/containers/ProjectContainer/TestcaseContainer/index.jsx]: changes committed 👍🏻

File: src/Components/ProjectsComponent/TestcaseComponents/components/AddTeststepsModal.jsx [hunk 7]
```diff
@@ -1,114 +1,122 @@
-import React, { useRef } from 'react';
-import PropTypes from 'prop-types';
-import { Col, Row } from 'react-bootstrap';
+import React, { useRef } from "react";
```
Comment (tanmaythole) [inline src/Components/ProjectsComponent/TestcaseComponents/components/AddTeststepsModal.jsx:1]: Hello @ChinmaySonawane , due to prettier multiple changes detected in this file

Comment (ChinmaySonawane) [reply src/Components/ProjectsComponent/TestcaseComponents/components/AddTeststepsModal.jsx:1]: okay

File: src/store/Execute/sagas.js [hunk 8]
```diff
@@ -1,37 +1,27 @@
...
-    } catch (error) {
-        toast.error(error.data.error)
+  try {
+    delete payload.data;
+    const response = yield call(executeTestcaseApi, payload);
+    yield put(addExecuteSuccess(response));
+  } catch (error) {
+    const err = error.data.error;
+    yield put(addExecuteFailure(err));
+    if (typeof err === "object") {
```
Comment (ChinmaySonawane) [inline src/store/Execute/sagas.js:17]: make sure err is never a array because typeof on array will also give you the object

**Review by ChinmaySonawane — APPROVED** [review]

File: src/store/Execute/sagas.js (hunk 8 above)
Comment (tanmaythole) [reply src/store/Execute/sagas.js:17]: custom error messages are added when the type of error differs

Comment (ChinmaySonawane) [reply src/store/Execute/sagas.js:17]: Okay

**Review by ChinmaySonawane — APPROVED** [review]

**Pull request merged**